### ISSUE TITLE
[A11y]: The sidebar menu should be audited

### ISSUE THREAD

**Issue by Sidnioulz** [storybookjs/storybook#32236]
### Describe the bug

Storybook has a sidebar menu that contains a mixture of global options (often with kb shortcuts) and navigation links. This mixed status makes me doubt whether we can apply a traditional menu role.

<img width="484" height="573" alt="Image" src="https://github.com/user-attachments/assets/312b4691-9692-401b-b945-44e5f323664d" />

The menu role should not contain navigation elements according to https://adrianroselli.com/2017/10/dont-use-aria-menu-roles-for-site-nav.html. APGs have been updated to provide the same warning: https://www.w3.org/WAI/ARIA/apg/patterns/menubar/examples/menubar-navigation/. This may not be applicable to us as we do have a majority of non-navigation elements.

Because it's not certain what we should do, and this concerns a single item, I'm treating this issue as low-priority. We could consider reworking the menu to make it fall more closely into a standard ARIA role.

### Reproduction link

Any storybook instance

### System

```bash
9.2 alpha (next), Node 22, Linux
```
